From 38f2cd041c029597a9b487cacb7e2c2aba40caa0 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Sat, 10 Aug 2024 03:19:52 +0000
Subject: [PATCH] dts: s32g: remove the #interrupt-cells property from non
 interrupt controller nodes

According to devicetree binding document for interrupts, the
"#interrupt-cells" should be a property of a interrupt-controller,
for non-interrupt-controller dts nodes, the "#interrupt-cells" should
not exist, or else the interrupt parse flow may be affacted and
undefined behaviours may happen.

So remove the "#interrupt-cells" property of non-interrupt-controller
for S32G platforms to fix irq parse errors, as following:
 hse-uio 40210000.mu0b: failed to register hse-mu0b-rx irq, line -517

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/s32cc.dtsi | 2 --
 arch/arm64/boot/dts/freescale/s32g.dtsi  | 1 -
 2 files changed, 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/s32cc.dtsi b/arch/arm64/boot/dts/freescale/s32cc.dtsi
index f94bb2ff532f..35cc892fb8b0 100644
--- a/arch/arm64/boot/dts/freescale/s32cc.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32cc.dtsi
@@ -200,7 +200,6 @@ soc {
 
 		rtc0: rtc@40060000 {
 			compatible = "nxp,s32cc-rtc";
-			#interrupt-cells = <3>;
 			reg = <0x0 0x40060000 0x0 0x1000>;
 			interrupts = <GIC_SPI 121 IRQ_TYPE_LEVEL_HIGH>;	// RTC irq - GIC 153
 			clocks = <&clks S32CC_SCMI_CLK_RTC_REG>,
@@ -597,7 +596,6 @@ hse: crypto {
 			compatible = "simple-bus";
 			#address-cells = <2>;
 			#size-cells = <2>;
-			#interrupt-cells = <3>;
 			memory-region = <&hse_reserved>;
 			ranges;
 
diff --git a/arch/arm64/boot/dts/freescale/s32g.dtsi b/arch/arm64/boot/dts/freescale/s32g.dtsi
index 53ea7ad23f09..0508ab860f47 100644
--- a/arch/arm64/boot/dts/freescale/s32g.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32g.dtsi
@@ -518,7 +518,6 @@ llce: llce@43a00000 {
 
 			#address-cells = <2>;
 			#size-cells = <2>;
-			#interrupt-cells = <3>;
 
 			llce_mb: llce_mb@43a00000 {
 				compatible = "nxp,s32g-llce-mailbox";
-- 
2.34.1

